src/bytes.rs

```diff
@@ -994,12 +1043,44 @@ unsafe fn promotable_to_vec(
     }
 }
 
+unsafe fn promotable_to_mut(
+    data: &AtomicPtr<()>,
+    ptr: *const u8,
+    len: usize,
+    f: fn(*mut ()) -> *mut u8,
+) -> BytesMut {
+    let shared = data.load(Ordering::Acquire);
+    let kind = shared as usize & KIND_MASK;
+
+    if kind == KIND_ARC {
+        shared_to_mut_impl(shared.cast(), ptr, len)
+    } else {
+        // The kind KIND_VEC always represents the full view of the underlying buffer,
+        // if it truncated it is first promoted to KIND_ARC. Thus we can reconstruct
+        // a Vec from it without leaking memory.
+        debug_assert_eq!(kind, KIND_VEC);
```

Comment:
This comment is wrong (but the code looks ok, like discussed elsewhere). It's possible to remove bytes from the beginning of the slice without promotion. However, the end of the view will always match the end of the buffer.

Reply:
I feel this is more of a native english speaking thing since truncated implies "at the end", but I will make it clearer

Reply:
Just to clarify, I have no issue with the word "truncated". My issue is with "full view of the underlying buffer".